src/helpers/supabase.ts

```diff
@@ -85,3 +85,32 @@ export const getRepoByGroupId = async (groupId: number) => {
     return "";
   }
 };
+
+export const bindGithubToTelegramUser = async (groupId: number, username: string, githubId: string) => {
+  const { data, error } = await supabase.from("telegram_users").upsert([
+    {
+      user_id: `${username}_${groupId}`,
+      github_id: githubId,
+      created_at: new Date().toUTCString(),
+      updated_at: new Date().toUTCString(),
+    },
+  ]);
+
+  if (error) {
+    console.error("Error adding/updating user:", error.message);
+    return null;
+  }
+
+  return data;
+};
+
+export const getUserGithubUsername = async (username: string, groupId: number) => {
+  const { data, error } = await supabase.from("telegram_users").select("github_id").eq("user_id", `${username}_${groupId}`);
```

Comment:
In the UbiquiBot database, we recently decided to primarily rely on the [GitHub user id](https://github.com/ubiquity/ubiquibot/issues/574) (in case people change their names.)

Perhaps it's more wise to do the same here? What do you think?

Reply:
> Perhaps it's more wise to do the same here? What do you think?

Its more API requests but should be worth it

would have to call `https://api.github.com/user/:username` after getting the `login` from the OAuth and then `https://api.github.com/user/:id` when there is a tagged user on the issue

I'd try it out and see if there is any permission issues

Reply:
Fixed here: [chore: use github id for oauth](https://github.com/ubiquity/telegram-ubiquibot/pull/17/commits/0e2e0c85e2733d5afbf5067379a16be0a3c8afbb) @pavlovcik